pyteal/ast/abi/transaction.py

```diff
@@ -0,0 +1,238 @@
+from enum import Enum
+from typing import TypeVar, Union, cast, List, Final
+from pyteal.ast.abi.type import BaseType, ComputedValue, TypeSpec
+from pyteal.ast.expr import Expr
+from pyteal.ast.int import Int
+from pyteal.ast.txn import TxnObject
+from pyteal.ast.gtxn import Gtxn
+from pyteal.types import TealType
+from pyteal.errors import TealInputError
+
+T = TypeVar("T", bound=BaseType)
+
+
+class TransactionType(Enum):
+    Transaction = "txn"
+    Payment = "pay"
+    KeyRegistration = "keyreg"
+    AssetConfig = "acfg"
+    AssetTransfer = "axfer"
+    AssetFreeze = "afrz"
+    ApplicationCall = "appl"
+
+
+TransactionType.__module__ = "pyteal"
+
+
+class TransactionTypeSpec(TypeSpec):
+    def __init__(self) -> None:
+        super().__init__()
+
+    def new_instance(self) -> "Transaction":
+        return Transaction()
+
+    def annotation_type(self) -> "type[Transaction]":
+        return Transaction
+
+    def is_dynamic(self) -> bool:
+        return False
+
+    def byte_length_static(self) -> int:
+        raise TealInputError("Transaction Types don't have a static size")
+
+    def storage_type(self) -> TealType:
+        return TealType.uint64
+
+    def __eq__(self, other: object) -> bool:
+        return type(self) is type(other)
+
+    def __str__(self) -> str:
+        return TransactionType.Transaction.value
+
+
+TransactionTypeSpec.__module__ = "pyteal"
+
+
+class Transaction(BaseType):
+    def __init__(self, spec: TransactionTypeSpec = TransactionTypeSpec()) -> None:
```

Comment:
nit: consider `= None` instead and creating the default value in the function body. There can be unintended side effects when you create a mutable object as a default value, since that value is reused for all function calls